gatsby/snippetUtils.ts

```diff
@@ -6,16 +6,33 @@ type Snippet = {
     snippet: string | undefined
 }
 
-export const resolveSnippetImports = (body: string, filePath: string): Snippet[] => {
-    const regex = /import\s+(.*?)\s+from\s+['"](.*?\/_snippets\/.*?)['"]/g
+export const resolveSnippetImports = (
+    body: string,
+    filePath: string,
+    restrictToOnlySnippetsDir: boolean = true
+): Snippet[] => {
+    const regex = restrictToOnlySnippetsDir
+        ? /import\s+(.*?)\s+from\s+['"](.*?\/_snippets\/.*?)['"]/g
+        : /import\s+(.*?)\s+from\s+['"](.*?\.mdx)['"]/g
     const imports: Snippet[] = []
     let match
     while ((match = regex.exec(body)) !== null) {
         const snippetPath = match[2]
         const cleanPath = path.resolve(path.dirname(filePath), snippetPath)
         if (fs.existsSync(cleanPath)) {
-            const snippet = fs.readFileSync(cleanPath, 'utf8').trimEnd()
-            imports.push({ component: match[1], path: match[2], snippet })
+            let snippet = fs.readFileSync(cleanPath, 'utf8').trimEnd()
+
+            // Recursively resolve nested imports using the snippet's own file path (for relative imports)
+            const nestedImports = resolveSnippetImports(snippet, cleanPath, false)
```

Comment:
I think this will work, but I think we don't need to add in the extra true false logic on only looking for snippets. Should just be any time we import md/mdx. I like your approach better.